### ISSUE TITLE
🐛 `cloudflared` does not reconnect after network outage

### ISSUE THREAD

**Issue by agneevX** [cloudflare/cloudflared#724]
**Describe the bug**

When my network goes down and comes back up, `cloudflared` does not appear to re-connect to Cloudflare servers automatically; the daemon must be manually restarted.

**To Reproduce**
1. Configure '...'
2. Run '....'
3. See error

**Expected behavior**

`cloudflared` should re-connect automatically.

**Environment and versions**
 - OS: Ubuntu
 - Architecture: arm64
 - Version: 2022.8.0

**Logs and errors**

```
2022-08-12T21:22:47Z ERR Unable to establish connection with Cloudflare edge error="DialContext error: dial tcp 198.41.200.33:7844: i/o timeout" connIndex=0 ip=198.41.200.33
2022-08-12T21:22:47Z ERR Serve tunnel error error="DialContext error: dial tcp 198.41.200.33:7844: i/o timeout" connIndex=0 ip=198.41.200.33
2022-08-12T21:22:47Z ERR Unable to establish connection with Cloudflare edge error="DialContext error: dial tcp 198.41.192.77:7844: i/o timeout" connIndex=3 ip=198.41.192.77
2022-08-12T21:22:47Z INF Retrying connection in up to 32s seconds connIndex=0 ip=198.41.200.33
2022-08-12T21:22:47Z ERR Unable to establish connection with Cloudflare edge error="DialContext error: dial tcp 198.41.192.47:7844: i/o timeout" connIndex=1 ip=198.41.192.47
2022-08-12T21:22:47Z ERR Serve tunnel error error="DialContext error: dial tcp 198.41.192.47:7844: i/o timeout" connIndex=1 ip=198.41.192.47
2022-08-12T21:22:47Z INF Retrying connection in up to 32s seconds connIndex=1 ip=198.41.192.47
2022-08-12T21:22:47Z ERR Unable to establish connection with Cloudflare edge error="DialContext error: dial tcp 198.41.200.43:7844: i/o timeout" connIndex=2 ip=198.41.200.43
2022-08-12T21:22:47Z ERR Serve tunnel error error="DialContext error: dial tcp 198.41.192.77:7844: i/o timeout" connIndex=3 ip=198.41.192.77
2022-08-12T21:22:47Z ERR Serve tunnel error error="DialContext error: dial tcp 198.41.200.43:7844: i/o timeout" connIndex=2 ip=198.41.200.43
2022-08-12T21:22:47Z INF Retrying connection in up to 32s seconds connIndex=2 ip=198.41.200.43
2022-08-12T21:22:47Z INF Retrying connection in up to 32s seconds connIndex=3 ip=198.41.192.77
```

**Additional context**

Sites are unavailable after internet comes back up:

![image](https://user-images.githubusercontent.com/19761269/184534851-146e4446-a2c5-408a-84fc-146ea1c1c029.png)

**EDIT**: add screenshot of unavailable site

**Comment by abelinkinbio** [comment]
Hi @agneevX - Thanks for reporting this. Are you able to give a bit more context on the cause and duration of the network outage as well as the behavior you noticed from cloudflared during and shortly after?

**Comment by agneevX** [comment]
Hello 👋!

The internet was out for ~8hrs. It's also fairly common for the network to be unavailable for similar periods of time. 

Upon reconnection, I receive Error 1033 for my sites, until the daemon is restarted. 

There's no log output when the internet is restored.

**Comment by sudarshan-reddy** [comment]
This has also been reported here:  https://github.com/cloudflare/cloudflared/issues/701 and internally so just letting you know we are looking into this!

**Comment by Albonycal** [comment]
could this be related to this https://github.com/cloudflare/cloudflared/issues/698 ?

**Comment by sudarshan-reddy** [comment]
They are different problems. We are working on a solution for that too.

**Comment by Wolfhound905** [comment]
I am also experiencing the same issue, thanks for the report @agneevX

**Comment by kdkasad** [comment]
Same issue here, although my instance of `cloudflared` is running inside a Docker container. After any network outages longer than a few hours, `cloudflared` won't reconnect.

**Comment by sudarshan-reddy** [comment]
We’ve identified the problem. A fix should come out shortly.

**Comment by sudarshan-reddy** [comment]
https://github.com/cloudflare/cloudflared/commit/e3390fcb15c52a8e2a33fd9002680852b12e2f67 will address this. There should be a release out this Monday. 

If you want to test it ahead of time, feel free to build this from source.

**Comment by agneevX** [comment]
Monday huh

**Comment by sudarshan-reddy** [comment]
You could still test it from the source.

**Comment by nmldiegues** [comment]
The release is currently blocked by another task, but we're working on unblocking it. Can't promise a date at this point, but it'll be as fast as we can.

**Comment by nmldiegues** [comment]
[Release 2022.8.4](https://github.com/cloudflare/cloudflared/commit/075ac1acf1b79c3dd5281d94188fa131b167483b) is out

Please give it a go and let us know.

**Comment by agneevX** [comment]
Thanks, I'll be able to provide an update soon.

**Comment by agneevX** [comment]
I can confirm that the issue is resolved. Much appreciated.
